server/delay_server/src/irodsDelayServer.cpp

```diff
@@ -674,6 +675,46 @@ namespace
 
         return {qstr, job};
     } // make_delay_queue_query_processor
+
+    bool is_current_server_leader() {
+        std::string hostname;
+        char hn[HOST_NAME_MAX + 1]{};
+
+        if (gethostname(hn, sizeof(hn)) != 0) {
+            logger::delay_server::warn("{}: Failed to retrieve local server's hostname.", __func__);
+            return false;
+        }
+
+        if (const auto hn_resolved = resolve_hostname(hn, hostname_resolution_scheme::match_preferred); hn_resolved) {
+            hostname = *hn_resolved;
+        }
+        else {
+            hostname = hn;
+        }
+        irods::experimental::client_connection conn;
+        GridConfigurationInput input{};
+        std::strcpy(input.name_space, "delay_server");
+        std::strcpy(input.option_name, "leader");
+
+        GridConfigurationOutput* output{};
+        irods::at_scope_exit free_output{[&output] { std::free(output); }};
+
+        if (const auto ec = rc_get_grid_configuration_value(static_cast<RcComm*>(conn), &input, &output); ec != 0) {
+            logger::delay_server::warn("{}: Failed to retrieve leader config option.", __func__);
+            return false;
+        }
+
+        std::string leader;
+        if (const auto hn_resolved = resolve_hostname(output->option_value, hostname_resolution_scheme::match_preferred); hn_resolved) {
+            leader = *hn_resolved;
+        }
+        else {
+            leader = output->option_value;
+        }
+
+        return hostname == leader;
```

Comment:
Maybe could avoid the extra string here like this:
```suggestion
        if (const auto hn_resolved = resolve_hostname(output->option_value, hostname_resolution_scheme::match_preferred); hn_resolved) {
            return hostname == *hn_resolved;
        }
        
        return hostname == output->option_value;
```
But, not as "self-documenting". Just a thought.